libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/SalesforceTLSSocketFactory.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2015-present, salesforce.com, inc.
+ * All rights reserved.
+ * Redistribution and use of this software in source and binary forms, with or
+ * without modification, are permitted provided that the following conditions
+ * are met:
+ * - Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * - Neither the name of salesforce.com, inc. nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.auth;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+
+/**
+ * Custom SSLSocketFactory implementation to disable the default usage of TLS 1.0, and use
+ * only TLS 1.1 or 1.2, as per Salesforce server's new requirements. This implementation is
+ * only required for Android 4.4 (KitKat), since TLS 1.1 or 1.2 are the default in Android
+ * Lollipop (5.0) and above.
+ *
+ * FIXME: Remove this class once minApi >= Lollipop.
+ *
+ * @author bhariharan
+ */
+public class SalesforceTLSSocketFactory extends SSLSocketFactory {
+
+    private static SalesforceTLSSocketFactory INSTANCE;
+    private SSLSocketFactory delegate; // XXX needs to be called delegate to work around okhttp3 bug: see https://github.com/square/okhttp/issues/2323#issuecomment-185925554
```

Comment:
Had to rename the private variable to delegate 
See https://github.com/square/okhttp/issues/2323#issuecomment-185925554